src/python/WMComponent/JobCreator/CreateWorkArea.py

```diff
@@ -35,35 +29,20 @@ def createDirectories(dirList):
     Create the directory if everything is sane
 
     """
-
-
-    #This is gonna be tricky
-    cmdList = []
-    cmdArgs = ['mkdir']
-
-    while len(dirList) > 500:
-        cmdArgs.extend(dirList[:500])
-        cmdList.append(cmdArgs)
+    for sdirList in grouper(dirList, 500):
         cmdArgs = ['mkdir']
-        dirList = dirList[500:]
-    if len(dirList) > 0:
-        cmdArgs.extend(dirList)
-        cmdList.append(cmdArgs)
-
-    logging.info('Executing makedir commands')
-    for command in cmdList:
-        pipe = Popen(command, stdout = PIPE, stderr = PIPE, shell = False)
+        cmdArgs.extend(sdirList)
+        pipe = Popen(cmdArgs, stdout = PIPE, stderr = PIPE, shell = False)
```

Comment:
Maybe. 1000 directories in under 100 ms, not sure it matters.

Sent from a mobile device. Please excuse my brevity or transcription errors.

On Feb 26, 2016, at 15:43, Alan Malta Rodrigues <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/python/WMComponent/JobCreator/CreateWorkArea.py<https://github.com/dmwm/WMCore/pull/6638#discussion_r54306183>:

>          cmdArgs = ['mkdir']
> -        dirList = dirList[500:]
> -    if len(dirList) > 0:
> -        cmdArgs.extend(dirList)
> -        cmdList.append(cmdArgs)
> -
> -    logging.info<http://logging.info>('Executing makedir commands')
> -    for command in cmdList:
> -        pipe = Popen(command, stdout = PIPE, stderr = PIPE, shell = False)
> +        cmdArgs.extend(sdirList)
> +        pipe = Popen(cmdArgs, stdout = PIPE, stderr = PIPE, shell = False)


Interesting, I did a quick script to time how long it takes to create 1k directories using 1) subprocess.call 2) os.mkdir and 3) subprocess.Popen. These were the results (ran 100 times each)
AlanWorkingBook:pythonTest amaltaro$ python Timer2.py
=> call function elasped time: ms 69.0529680252
=> mkdir function elasped time: ms 79.0190982819
=> Popen function elasped time: ms 30.5561995506

So, popen seems to be our best pick for this case ;)

-
Reply to this email directly or view it on GitHub<https://github.com/dmwm/WMCore/pull/6638/files#r54306183>.